lib/util/result.ts

```diff
@@ -1,73 +1,469 @@
+/* eslint-disable promise/no-nesting */
+import { logger } from '../logger';
+
 interface Ok<T> {
-  readonly success: true;
-  readonly value: T;
+  readonly ok: true;
+  readonly val: NonNullable<T>;
+  readonly err?: never;
 }
 
 interface Err<E> {
-  readonly success: false;
-  readonly error: E;
+  readonly ok: false;
+  readonly err: NonNullable<E>;
+  readonly val?: never;
+  readonly _uncaught?: true;
```

Comment:
what's this?

Reply:
The errors during transforms should be explicitly caught and handled, we don't want surprises to happen there (bad for types). To enforce this, we mark uncaught errors and re-throw them on `unwrap()` call.

Reply:
can you add a small tsdoc to mark this as internal use?